src/activity/activity.service.ts

```diff
@@ -19,15 +19,18 @@ export class ActivityService {
     return activities;
   }
 
-  findOne(id: number) {
-    return `This action returns a #${id} activity`;
+  async findOne(id: string) {
+    const activity = await this.activityModel.findById(id).exec();
+    return activity;
   }
 
-  update(id: number, updateActivityDto: UpdateActivityDto) {
-    return `This action updates a #${id} activity`;
+  update(id: string, updateActivityDto: UpdateActivityDto) {
```

Comment:
I see that all methods of `user.service.ts` in the project use `async/await`.
However, there is no addition here. I suggest that they should be unified, but I think you probably forgot to add it.

Reply:
> I see that all methods of `user.service.ts` in the project use `async/await`. However, there is no addition here. I suggest that they should be unified, but I think you probably forgot to add it.

We use `async/await` only when the function needs `await` somewhere, the update function doesn't have any await now, so I think it's better to keep it a sync function.